### PR TITLE
Devon

### PR DESCRIPTION
**Pull request by Laddicus**
Added gallery page (html, css, python, js)

Allowed for javascript files to be added

### CODE REVIEW

File: nplbam/app/gallery.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+from flask import Blueprint, redirect, render_template, request, current_app, url_for
+from flask import session as flask_session
+from sqlalchemy.orm import sessionmaker
+
+from .db import db
+
+import os
+
+import nplbam.app.file_downloads as file_downloads
```
Comment (ValliereMagic) [inline nplbam/app/gallery.py]: Unused import?

File: nplbam/app/gallery.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        images.append(x.fileName)
+
+    # Close the database like a good boy
+    db_session.close()
+
+    # Render the page
+    return render_template("gallery.html", animalID=viewID, title="View {}".format(animal_entry.name), info=info, images=images)
+
+# Route to view animal page.
+@bp.route("/animal_viewed", methods=['POST', 'GET'])
```
Comment (ValliereMagic) [inline nplbam/app/gallery.py]: Unnecessary function. This functionality already exists, and is overwriting existing routes.

File: nplbam/app/gallery.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
+import json
+from datetime import date
+
+from flask import Blueprint, redirect, render_template, request, current_app, url_for
```
Comment (ValliereMagic) [inline nplbam/app/gallery.py]: Are you using current_app? Or url_for?

File: nplbam/app/gallery.py [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+bp = Blueprint('gallery', __name__, url_prefix="")
+
+
+# Route to view animal page.
+@bp.route("/gallery")
+def gallery():
+    """
+    Page to view images and and limited information associated with animal
+    """
+    # Check if they are logged in
```
Comment (ValliereMagic) [inline nplbam/app/gallery.py:19]: Will need to add appropriate access role restrictions

File: nplbam/app/gallery.py [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+from flask import Blueprint, redirect, render_template, request, current_app, url_for
+from flask import session as flask_session
+from sqlalchemy.orm import sessionmaker
+
+from .db import db
+
+import os
+
+import nplbam.app.file_downloads as file_downloads
+
```
Comment (ValliereMagic) [inline nplbam/app/gallery.py]: In other words, please cleanup unused imports.

File: nplbam/app/templates/edit_account.html [hunk 6]
```diff
@@ -21,10 +21,10 @@
             </div>
             <div class="item">
                 <label for="userLVL">User Access Level: </label>
+                <input type="number" id="userLVL" value="{{ info.get("user_lvl") }}" name="userLVL" />
```
Comment (ValliereMagic) [inline nplbam/app/templates/edit_account.html:24]: Didn't you move this to the bottom before?

**Review by ValliereMagic — CHANGES_REQUESTED** [review]

File: nplbam/app/templates/edit_account.html (hunk 6 above)
Comment (ValliereMagic) [reply nplbam/app/templates/edit_account.html:24]: or was that from a different file?

Comment (Laddicus) [reply nplbam/app/templates/edit_account.html:24]: That was the new_account page this change is to match it

File: nplbam/app/gallery.py [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+    # Get a dictionary for required information
+    info = {}
+
+    info["name"] = animal_entry.name
+    info["stage"] = animal_entry.stage
+    info["days"] = (date.today() - animal_entry.stageDate).days
+
+    for x in animal_entry.textAnswers:
+        info[x.questionName] = x.answer
```
Comment (NevasPrime) [inline nplbam/app/gallery.py:43]: Why are you grabbing textAnswers if they're never used?

**Review by NevasPrime — COMMENTED** [review]
Honestly the gallery page is missing a lot. Such as stages, but I guess i'll do that this coming week.

File: nplbam/app/gallery.py (hunk 7 above)
Comment (Laddicus) [reply nplbam/app/gallery.py:43]: The notes are used on the gallery page

**Review by NevasPrime — APPROVED** [review]

**Review by ValliereMagic — APPROVED** [review]

**Pull request merged**